app/workers/irker_worker.rb

```diff
@@ -0,0 +1,147 @@
+require 'json'
+require 'socket'
+
+class IrkerWorker
+  include Sidekiq::Worker
+
+  def perform(project_id, chans, colors, push_data)
+    project = Project.find(project_id)
+
+    # Get config parameters
+    irker_server, irker_port = irker_daemon_config
+
+    return false unless setup_members chans, irker_server, irker_port
+
+    repo_name = push_data['repository']['name']
+    committer = push_data['user_name']
+    branch    = push_data['ref'].gsub(%r'refs/[^/]*/', '')
+
+    if colors
+      repo_name = "\x0304#{repo_name}\x0f"
+      branch    = "\x0305#{branch}\x0f"
+    end
+
+    # Firsts messages are for branch creation/deletion
+    send_branch_updates push_data, project, repo_name, committer, branch
+
+    # Next messages are for commits
+    send_commits push_data, project, repo_name, committer, branch, colors
+
+    @socket.close
```

Comment:
At this point the `@socket` feels a bit like a magic variable. What about putting it into a method? i.e. `close_connection`